protostar/testing/environments/cairo_test_execution_environment.py

```diff
@@ -0,0 +1,80 @@
+import asyncio
+from typing import Any, Optional
+
+from starkware.cairo.common.cairo_function_runner import CairoFunctionRunner
+from starkware.cairo.lang.compiler.program import Program
+from starkware.cairo.lang.vm.vm_exceptions import VmException
+
+from protostar.testing.environments.cairo_test_cheatcode_factory import (
+    CairoTestCheatcodeFactory,
+)
+from protostar.testing.environments.test_execution_environment import (
+    TestExecutionEnvironment,
+    TestExecutionResult,
+)
+from protostar.testing.starkware.execution_resources_summary import (
+    ExecutionResourcesSummary,
+)
+from protostar.testing.test_environment_exceptions import RevertableException
+
+
+class CairoTestExecutionEnvironment(TestExecutionEnvironment):
+    def __init__(self, program: Program, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self._program = program
+        self._max_steps = self.state.config.max_steps
+        self._profiling = self.state.config.profiling
+
+    async def execute(self, function_name: str) -> TestExecutionResult:
+        assert (
+            not self.state.config.profiling
+        ), "Profiling is not supported yet"  # TODO: Plug in profiling
```

Comment:
I think it will not be supported in this flow in the way it was before (and it is not going to work without tweaks). I am pro isolating profiler as a separate library soon and making it a cheatcode instead.

Reply:
Are we deprecating it or just not minding that now at all?

Reply:
not minding that now at all IMO, it still works for old protostar